lib/model/Mode.js

```diff
@@ -0,0 +1,55 @@
+const Physical = require(path.join(__dirname, 'Physical.js'));
+
+module.exports = class Mode {
+  constructor(jsonObject, fixture) {
+    this.jsonObject = jsonObject; // calls the setter
+    this.fixture = fixture; // also calls the setter
+  }
+
+  set jsonObject(jsonObject) {
+    this._jsonObject = jsonObject;
+    this._cache = {};
+  }
+
+  set fixture(fixture) {
+    this._fixture = fixture;
+    this._cache = {};
+  }
+
+
+  get name() {
+    return this._jsonObject.name; // required
+  }
+
+  get shortName() {
+    return this._jsonObject.shortName || this._jsonObject.name;
+  }
+
+  get physicalOverride() {
+    if (!('physicalOverride' in this._cache)) {
+      this._cache.physicalOverride = 'physical' in this._jsonObject ? new Physical(this._jsonObject.physical) : null;
+    }
+    return this._cache.physicalOverride;
+  }
+
+  // fixture's physical with mode's phyiscal override applied on
```

Comment:
But, what happens if a sub-value (e.g. `bulb.colorTemperature`) is set in only one of both physical sections and another value in the same category (e.g. `bulb.type`) is set in the other?